components/Graph.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useState, useEffect } from 'react';
+import { View, TouchableOpacity, Dimensions } from 'react-native';
+import { LineChart } from 'react-native-gifted-charts';
+import { FText } from '~/components/Text/FText';
+
+interface DataPoint {
+  value: number;
+  label: string;
+}
+
+const Graph = () => {
+  // Generate mock data only once
+  // const [allData] = useState<DataPoint[]>(
+  //   Array.from({ length: 365 }, (_, i) => ({
+  //     value: Math.floor(Math.random() * (2763 - 1000 + 1)) + 1000, // Random price
+  //     label: new Date(Date.now() - i * 24 * 60 * 60 * 1000).toISOString(), // ISO date
+  //   }))
+  // );
+
+  const [allData] = useState<DataPoint[]>([]);
+
+  const [selectedRange, setSelectedRange] = useState<string>('1month');
+  const [filteredData, setFilteredData] = useState<DataPoint[]>([]);
+  const [currentValue, setCurrentValue] = useState<number>(0);
+  const [currentDate, setCurrentDate] = useState<string>('');
+  const [containerWidth, setContainerWidth] = useState<number>(
+    Dimensions.get('window').width - 32 // Default width with padding
+  );
+
+  // Filter data based on selected range
+  useEffect(() => {
+    const now = new Date();
+    let range = 0;
+
+    switch (selectedRange) {
+      case '1day':
+        range = 1;
+        break;
+      case '1week':
+        range = 7;
+        break;
+      case '1month':
+        range = 30;
+        break;
+      case '1year':
+        setFilteredData([...allData].reverse()); // Reverse the order for latest data on the right
+        return;
+    }
+
+    const newFilteredData = allData
+      .filter((d) => new Date(d.label!) >= new Date(now.getTime() - range * 24 * 60 * 60 * 1000))
+      .reverse(); // Reverse the filtered data
+    setFilteredData(newFilteredData);
+  }, [selectedRange, allData]);
+
+  // Reset to last data point
+  useEffect(() => {
+    if (filteredData.length > 0) {
+      const lastData = filteredData[filteredData.length - 1];
+      setCurrentValue(lastData.value);
+      setCurrentDate(new Date(lastData.label!).toLocaleDateString());
+    }
+  }, [filteredData]);
+
+  const handlePointer = ({ pointerIndex }: { pointerIndex: number }) => {
+    if (pointerIndex === -1) {
+      // Reset to the last data point when pointer is released
+      const lastData = filteredData[filteredData.length - 1];
+      setCurrentValue(lastData?.value || 0);
+      setCurrentDate(new Date(lastData?.label || '').toLocaleDateString());
```

Comment:
So did you manage to fix this in the end?

Reply:
No but i kept it as it is used when you switch time frames, the pointerIndex is == -1 when switching time frames so it is usefull to not still have the last selected value in a different time frame (e.g. a value from 3 months ago when switching from yearly to monthly).